spark-pipeline-engine/src/main/scala/com/acxiom/pipeline/applications/ApplicationUtils.scala

```diff
@@ -76,10 +78,21 @@ object ApplicationUtils {
     val globalPipelineParameters = generatePipelineParameters(application.pipelineParameters, Some(PipelineParameters()))
     val ctx = pipelineExecution.pipelineContext
       .copy(globals = generateGlobals(execution.globals, rootGlobals.get, defaultGlobals))
-        .copy(parameters = generatePipelineParameters(execution.pipelineParameters, globalPipelineParameters).get)
+      .copy(parameters = generatePipelineParameters(execution.pipelineParameters, globalPipelineParameters).get)
     pipelineExecution.asInstanceOf[DefaultPipelineExecution].copy(pipelineContext = ctx)
   }
 
+  private def generatePipelines(execution: Execution, application: Application): List[Pipeline] = {
+    if (execution.pipelines.isEmpty && execution.pipelineIds.isEmpty) {
+      throw new IllegalArgumentException("Either pipelines or pipelineIds must be provided for an execution")
+    }
+    val executionPipelines = execution.pipelines.getOrElse(List[DefaultPipeline]())
+    val pipelineIds = execution.pipelineIds.getOrElse(List[String]())
+    val applicationPipelines = application.pipelines.getOrElse(List[DefaultPipeline]())
```

Comment:
The pipelineIds element should be used to drive the pipelines for the execution. Each id should check to see if the execution.pipelines array has the definition first and then check the application pipelines array. If the pipelineIds array does not exist or is empty, then use the pipelines array of the execution as the pipelines. If the executions array is empty or not defined, then use the application pipelines array. Finally throw an exception is nothing is defined.